libraries/chain/include/eosio/chain/webassembly/interface.hpp

```diff
@@ -338,6 +338,20 @@ namespace webassembly {
          */
          int32_t recover_key(legacy_ptr<const fc::sha256> digest, legacy_span<const char> sig, legacy_span<char> pub) const;
 
+         /**
+         * Calculates the public key used for a given signature on a given digest.
+         * Unlike @ref recover_key it returns an error code rather than throws an exception with invalid input.
+         *
+         * @ingroup crypto
+         * @param digest - digest of the message that was signed.
+         * @param sig - signature.
+         * @param[out] pub - pointer to output buffer for the public key result.
+         * @param[inout] publen - pointer to an integer initially set to the expected size of output, and overwritten by the written size
```

Comment:
@param[out] publen - pointer to the size of output buffer of the public key `pub`. This value can be used when `error_code::insufficient_output_buffer` is returned to determine required size of `pub`.

Reply:
``` c++
int recover_key_safe(const capi_checksum256* digest, const char* sig, size_t sigsize,
  char* pub,
  size_t* publen);

// after review
int recover_key_safe(const capi_checksum256* digest, const char* sig, size_t sigsize,
  char* pub,
  size_t pubsize,
  size_t* publen);
```
In that case, is the above function signature for intrinsic right? AFAIK, `span<char>` is translated into `{char*,size_t}`. I considered reusing the last argument `size_t* publen` for the size of the given output buffer to store written size.

Reply:
This looks good.